### PR TITLE
[MOS-395] Fix issue with inability to send SMS

### PR DESCRIPTION
**Pull request by dawidwojtas-mudita**
Get rid of unnecessary separators which can be
problematic for cellular modem. Add modem soft
reset if send SMS message will fail.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [ ] Has unit tests if possible
- [ ] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

**Review by kkleczkowski — APPROVED** [review]

**Review by Maciej-Mudita — APPROVED** [review]

File: module-utils/phonenumber/PhoneNumber.cpp [hunk 1]
```diff
@@ -30,15 +30,17 @@ PhoneNumber::PhoneNumber(const std::string &phoneNumber, country::Id defaultCoun
     : countryCode(defaultCountryCode)
 {
     auto &util = *phn_util::GetInstance();
-    util.ParseAndKeepRawInput(phoneNumber, country::getAlpha2Code(countryCode), &pbNumber);
+    auto number = phoneNumber;
+    number.erase(std::remove_if(number.begin(), number.end(), &PhoneNumber::CharacterToRemove), number.end());
```
Comment (Lefucjusz) [inline module-utils/phonenumber/PhoneNumber.cpp:34]: You don't have to use `&` when dealing with pointer to function ;)

File: pure_changelog.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 * Implemented partial refresh of the display
 
 ### Fixed
+* Fixed SMS messages that can't send
```
Comment (Lefucjusz) [inline pure_changelog.md:12]:
```suggestion
* Fixed issue with inability to send SMS
```

**Review by Lefucjusz — APPROVED** [review]

File: module-utils/phonenumber/PhoneNumber.cpp (hunk 1 above)
Comment (dawidwojtas-mudita) [reply module-utils/phonenumber/PhoneNumber.cpp]: Done

File: pure_changelog.md (hunk 2 above)
Comment (dawidwojtas-mudita) [reply pure_changelog.md]: Done :)

**Pull request merged**